src/hooks/useDebounce.ts

```diff
@@ -0,0 +1,25 @@
+import { useState, useEffect } from 'react';
+
+function useDebounce<T>(
+  initialValue: T,
+  time: number
+): [T, T, React.Dispatch<T>] {
+  const [value, setValue] = useState<T>(initialValue);
+  const [debouncedValue, setDebouncedValue] = useState<T>(initialValue);
+
+  useEffect(() => {
+    const debounce = setTimeout(() => {
+      setDebouncedValue(value)
+    }, time);
+
+    return () => {
+      clearTimeout(debounce);
+    }
```

Comment:
IMO it's really good to debounce inputs, but do we actually need additional useState?

Reply:
I think it would be better if you could compare "throttling" with "debounce". Let's study together next time

Reply:
@jbeat30 Most definitely agree! I believe both of them appear quite frequently in interviews

Reply:
@devjiwonchoi the whole purpose of using double state was to compare the incoming value and the value to debounce - will honestly have to look at a couple more tutorials to reduce it to single state I guess?

Reply:
AFAIU

throttle: delay execution for a certain amount of time during the event
debounce: delay execution until the event ends